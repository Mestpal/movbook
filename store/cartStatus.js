export const state = {
  cartItems: []
}

export const getters = {
  cartItems: (state) => { return state.cartItems }
}

export const mutations = {
  UPDATE_CART (state, newItem) {
    console.log('newItem')
    state.cartItems.push(newItem)
  },
  REMOVE_CONTENT_BY_ID (state, id) {
    let found = false
    state.cartItems.forEach((item, index) => {
      if (item.id === id && !found) {
        state.cartItems.splice(index, 1)
        found = true
      }
    })
  }
}

export const actions = {
  addItemToCart ({ commit }, newItem) {
    console.log('newItem')
    commit('UPDATE_CART', newItem)
  },
  deleteContent ({ commit }, id) {
    commit('REMOVE_CONTENT_BY_ID', id)
  }
}
