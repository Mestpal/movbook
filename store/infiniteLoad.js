export const state = {
  page: 1
}

export const getters = {
  page: (state) => { return state.page }
}

export const mutations = {
  UPDATE_PAGE (state, newPage) {
    state.page = newPage
  }
}

export const actions = {
  updatePage ({ commit }, newPage) {
    commit('UPDATE_PAGE', newPage)
  }
}
