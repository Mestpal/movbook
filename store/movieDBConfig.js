import config from '@/config/api'

export const state = {
  configuration: {}
}

export const getters = {
  configuration: (state) => { return state.configuration }
}

export const mutations = {
  UPDATE_CONFIGURATION (state, newConfiguration) {
    state.configuration = newConfiguration
  }
}

export const actions = {
  async getMovieDBConfig ({ commit }) {
    const response = await this.$axios.get(`${config.api.url}/configuration?api_key=${config.api.apikey}`)
    commit('UPDATE_CONFIGURATION', response.data)
  }
}
