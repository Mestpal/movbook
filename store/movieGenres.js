export const state = {
  genres: [],
  selectedGenre: {}
}

export const getters = {
  genres: (state) => { return state.genres },
  selectedGenre: (state) => { return state.selectedGenre }
}

export const mutations = {
  UPDATE_GENRES (state, newGenres) {
    state.genres = newGenres
  },
  UPDATE_SELECTED_GENRE (state, selectedGenre) {
    state.selectedGenre = selectedGenre
  }
}

export const actions = {
  async getFilmGenres ({ commit }) {
    const response = await this.$axios.get('https://api.themoviedb.org/3/genre/movie/list?api_key=059b24ced08b9601d550dc1bda480265&language=en-US')
    commit('UPDATE_GENRES', response.data.genres)
  },
  updateSelectedGenre ({ commit }, data) {
    if (data.id && data.name) {
      commit('UPDATE_SELECTED_GENRE', data)
    }
  }
}
