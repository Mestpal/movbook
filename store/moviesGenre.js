import config from '@/config/api'

export const state = {
  genres: [],
  selectedGenre: {}
}

export const getters = {
  genres: (state) => { return state.genres },
  selectedGenre: (state) => { return state.selectedGenre }
}

export const mutations = {
  UPDATE_GENRES (state, newGenres) {
    state.genres = newGenres
  },
  UPDATE_SELECTED_GENRE (state, selectedGenre) {
    state.selectedGenre = selectedGenre
  }
}

export const actions = {
  async getFilmGenres ({ commit }) {
    const response = await this.$axios.get(`${config.api.url}/genre/movie/list?api_key=${config.api.apikey}&language=en-US`)
    commit('UPDATE_GENRES', response.data.genres)
  },
  updateSelectedGenre ({ commit }, data) {
    if (data.id && data.name) {
      commit('UPDATE_SELECTED_GENRE', data)
    }
  }
}
