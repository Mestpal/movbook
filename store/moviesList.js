export const state = {
  moviesList: []
}

export const getters = {
  moviesList: (state) => { return state.moviesList }
}

export const mutations = {
  UPDATE_MOVIE_LIST (state, newMovies) {
    state.moviesList = state.moviesList.concat(newMovies)
  },
  RESET (state) {
    state.moviesList = []
  }
}

export const actions = {
  async getMoviesList ({ commit }, data) {
    const response = await this.$axios.get(`https://api.themoviedb.org/3/discover/movie?api_key=059b24ced08b9601d550dc1bda480265&language=en-US&sort_by=original_title.asc&include_adult=false&include_video=false&page=${data.page}&with_genres=${data.genreId}`)
    commit('UPDATE_MOVIE_LIST', response.data.results)
  },
  async getMoviesListLive ({ commit }, data) {
    const response = await this.$axios.get(`https://api.themoviedb.org/3/search/movie?api_key=059b24ced08b9601d550dc1bda480265&language=en-US&query=${data.query}&page=1&include_adult=false`)
    commit('UPDATE_MOVIE_LIST', response.data.results)
  },
  resetMovieListData ({ commit }) {
    commit('RESET')
  }
}
