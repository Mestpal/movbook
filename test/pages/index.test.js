import Vue from 'vue'
import Vuetify from 'vuetify'
import Vuex from 'vuex'
import { shallowMount, createLocalVue } from '@vue/test-utils'

import index from '@/pages/index'
import { gettersDBC } from '@/store/movieDBConfig'
import { gettersG } from '@/store/moviesGenre'

Vue.use(Vuetify)
Vue.use(Vuex)
const localVue = createLocalVue()

describe('index.vue page', () => {
  let wrapper
  const genreList = [
    {
      id: 16,
      name: 'animation'
    }
  ]

  beforeEach(() => {
    wrapper = shallowMount(index, {
      localVue,
      computed: {
        configuration () {
          return {
            images: {
              base_url: 'test_base_url/'
            }
          }
        },
        genres () {
          return genreList
        }
      },
      getters: {
        gettersG,
        gettersDBC
      },
      methods: {
        checkGenresStatus () {
          return ''
        },
        checkConfiguration () {
          return ''
        },
        resetMovieListData () {
          return ''
        }
      }
    })
  })

  it('reders the page', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('checkGenresStatus not need to obtain films list', () => {
    wrapper.vm.getFilmGenres = jest.fn()
    wrapper.vm.checkGenresStatus()
    expect(wrapper.vm.getFilmGenres).not.toHaveBeenCalled()
  })
})
