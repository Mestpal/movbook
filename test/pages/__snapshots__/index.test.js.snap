// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index.vue page reders the page 1`] = `
"<div>
  <category-stub data=\\"[object Object]\\"></category-stub>
</div>"
`;
